aiosmtpd/tests/test_smtp.py

```diff
@@ -1518,3 +1519,46 @@ def test_timeout(self, plain_controller, client):
         time.sleep(0.1 + TimeoutController.Delay)
         with pytest.raises(SMTPServerDisconnected):
             client.mail("anne@example.com")
+
+
+class TestAuthArgs:
+    def test_warn_authreqnotls(self, caplog):
+        with pytest.warns(UserWarning) as record:
+            _ = Server(Sink(), auth_required=True, auth_require_tls=False)
+        assert len(record) == 1
+        assert (
+            record[0].message.args[0]
+            == "Requiring AUTH while not requiring TLS can lead to "
+               "security vulnerabilities!"
+        )
+        assert caplog.record_tuples[0] == (
+            "mail.log",
+            logging.WARNING,
+            "auth_required == True but auth_require_tls == False",
+        )
+
+    def test_log_authmechanisms(self, caplog):
```

Comment:
This test case is problematic... if I run it alone, it works. If I run the whole `test_smtp.py` file, it works.

But if I run it inside tox, it fails. What the...

Reply:
Okay, narrowed down the issue: That test fails _only_ if `test_main.py` runs before it.

Still scratching my head, but at least I can now narrow my focus.

Reply:
Narrowed down the culprit further. Something in `test_main.py::TestMain::test_debug_*` tests are messing up.

I'll fix it and push a fixed commit.

Reply:
Okay fixed. Apparently I don't need to save logging level and restore after test case is running.